libraries/apollo-ast/src/main/kotlin/com/apollographql/apollo3/ast/check_key_fields.kt

```diff
@@ -17,15 +17,15 @@ private class CheckKeyFieldsScope(
 
 fun checkKeyFields(operation: GQLOperationDefinition, schema: Schema, allFragmentDefinitions: Map<String, GQLFragmentDefinition>) {
   val parentType = operation.rootTypeDefinition(schema)!!.name
-  CheckKeyFieldsScope(schema, allFragmentDefinitions).checkField("Operation(${operation.name})", operation.selectionSet.selections, parentType)
+  CheckKeyFieldsScope(schema, allFragmentDefinitions).checkField("Operation(${operation.name})", operation.selections, parentType)
```

Comment:
This is a breaking change. `GQLSelectionSet` and `GQLArguments` are removed. There are a lot of places in the spec where there are list of symbols surrounded by `{ }` or `()`. Sometimes we had a node for them (like `GQLSelectionSet` in `GQLField.selectionSet`) but sometimes not (like `GQLObjectTypeDefinition.fields`).

Removing the intermediate node makes the callsite cleaner (like here) but looses the `sourceLocation` for the opening brace/parenthesis. We could also have gone the other direction and add extra intermediate nodes everywhere so it's all a tradeoff.

I went with "smaller/cleaner callsites" because this is what makes sense here I think as apollo-compiler doesn't really care about the position of the braces/parenthesis. Also we're already stripping whitespace and comments so it felt somewhat consistent.